kirby/api/ext.py

```diff
@@ -70,25 +85,48 @@ def init_kafka(self):
             self._consumer = KafkaConsumer(
                 self.name,
                 group_id=self.kirby_app.ctx.PACKAGE_NAME,
-                value_deserializer=lambda x: msgpack.loads(x, raw=False),
+                value_deserializer=kirby_value_deserializer,
                 **kafka_args,
             )
+            self._consumer.poll()
             self._producer = KafkaProducer(
-                value_serializer=msgpack.dumps, **kafka_args
+                value_serializer=kirby_value_serializer, **kafka_args
+            )
+
+    @staticmethod
+    def format_headers(headers):
+        if isinstance(headers, dict):
+            headers = list(headers.items())
+
+        if isinstance(headers, list):
+            return [
+                (header[0], kirby_value_serializer(header[1]))
+                for header in headers
+            ]
+        else:
+            raise RuntimeError(
+                f"The format of given headers ({headers}) is not correct "
+                "it must be a list of tuple or a dictionary"
```

Comment:
why allowing a list of tuples in the first place? a dict looks like a more suited container due to the uniqueness of the keys, no?

Reply:
Because in the first place Kafka was using a list of tuples. Thing that didn't seem convenient to use for us, so I wanted to extend the behavior to a dict.
I will not allow lists of tuples and only allow a dict for the reason you mention.